### ISSUE TITLE
block zero analysis needs reprocessing step

### ISSUE THREAD

**Issue by jtmoon79** [jtmoon79/super-speedy-syslog-searcher#74]
**_tl;dr_** blockzero analysis may find multiple `DateTimeParseInstr` patterns and must reparse the file if it does

### Problem

Given this log file

```text
$ head /var/log/system.log
Jan 10 18:33:00 hostname  newsyslog[94988]: logfile turned over due to size>500K
<38>1 2023-01-10T18:33:00.055834-08:00 hostname  sshguard 32449 - - Exiting on signal.
<38>1 2023-01-10T18:33:00.229544-08:00 hostname  sshguard 96545 - - Now monitoring attacks.
```

The service `newsyslog` is prepending message `logfile turned over due to size>500K` and using a different datetime format.

`s4` finds the first line to be date _1972-01-11T02:33:00_
```text
s4 -u /var/log/system.log | head
19720111T023300.000+0000:Jan 10 18:33:00 hostname  newsyslog[94988]: logfile turned over due to size>500K
20230111T023300.055+0000:<38>1 2023-01-10T18:33:00.055834-08:00 hostname  sshguard 32449 - - Exiting on signal.
20230111T023300.229+0000:<38>1 2023-01-10T18:33:00.229544-08:00 hostname  sshguard 96545 - - Now monitoring attacks.
```

### Solution

After the final `DateTimeParseInstr` is chosen, any `Sysline`s processed with any other `DateTimeParseInstr` must be reprocessed (the entire `Block` would need to be reprocessed).

In other words, if _any_ currently stored `Sysline` was processed using a different `DateTimeParseInstr` than all of the `Sysline`s must be removed (do not `drop` them), and then the entire zero `Block` must be reprocessed for `Sysline`s (luckily, should be fast as the `Block` and all `Line`s are still valid). Similar to what is done for reprocessing for missing years.

In this particular case, this is somewhat tricky as it is the very first line in the file with a differing datetime format. An ongoing presumption has been 
- the datetime stamp of a `Sysline` is the first line is the first line of the `Sysline`
so `Line`s in the beginning of the file without a "parent" `Sysline` will get dropped.

**Comment by jtmoon79** [comment]
This particular file causes an infinite loop for release builds. Debug breaks on a debug assert.
